UPtables/lib/asp/constraints/hard/time_windows.lp

```diff
@@ -0,0 +1,7 @@
+%%% Time windows: If there are time windos for a specific curriculum 
+%%% and a specific semester and the curriculum has a mandatory lecture,
+%%% recommended for the specific semester, the lecture must be scheduled 
+%%% within one time window
+
+:- assigned(C,WD,TF), course_component(C,_,_,_,_,lecture), mandatory(C,Cu), recommendation(C,S,Cu),
```

Comment:
@felixKubicek not quite sure anymore: Does the rule really enforces that the course has to be mandatory for our student teachers?